File: backend/models/stock.go
package models

import "time"

type Stock struct {
	ID          uint       `gorm:"primaryKey" json:"ID"`
	ItemID      uint       `gorm:"not null" json:"item_id"`
	WarehouseID uint       `gorm:"not null" json:"warehouse_id"`
	Quantity    float64    `gorm:"default:0" json:"quantity"`
	Item        Item       `json:"Item"`
	Warehouse   Warehouse  `json:"Warehouse"`
	StockLogs   []StockLog `json:"-"`
	CreatedAt   time.Time  `json:"CreatedAt"`
	UpdatedAt   time.Time  `json:"UpdatedAt"`
}

File: frontend/features/inventory/models/stock.ts
export interface Stock {
  ID: number;
  item_id: number;
  warehouse_id: number;
  quantity: number;
  Item: { ID: number; name: string; category: string; unit: string };
  Warehouse: { ID: number; name: string; location: string };
  CreatedAt: string;
  UpdatedAt: string;
}
